services/api-server/src/simcore_service_api_server/api/routes/solvers_jobs.py

```diff
@@ -271,3 +276,54 @@ async def get_job_outputs(
 
     job_outputs = JobOutputs(job_id=job_id, results=results)
     return job_outputs
+
+
+@router.get(
+    "/{solver_key:path}/releases/{version}/jobs/{job_id:uuid}/outputs/logfile",
+    response_class=RedirectResponse,
+    responses={
+        status.HTTP_200_OK: {
+            "content": {
+                "application/octet-stream": {
+                    "schema": {"type": "string", "format": "binary"}
+                },
+                "application/zip": {"schema": {"type": "string", "format": "binary"}},
+                "text/plain": {"schema": {"type": "string"}},
+            },
+            "description": "Returns a log file",
+        },
+        status.HTTP_404_NOT_FOUND: {"description": "Log not found"},
+    },
+)
+async def get_job_output_logfile(
+    solver_key: SolverKeyId,
+    version: VersionStr,
+    job_id: UUID,
+    user_id: PositiveInt = Depends(get_current_user_id),
+    director2_api: DirectorV2Api = Depends(get_api_client(DirectorV2Api)),
+):
+    """Special extra output with persistent logs file for the solver run.
+
+    NOTE: this is not a log stream but a predefined output that is only
+    available after the job is done.
+    """
+
+    logs_urls: dict[NodeName, DownloadLink] = await director2_api.get_computation_logs(
+        user_id=user_id, project_id=job_id
+    )
+
+    # if more than one node? should rezip all of them??
+    assert len(logs_urls) <= 1, "Current version only supports one node per solver"
+    for presigned_download_link in logs_urls.values():
+        logger.info(
+            "Redirecting '%s' to %s ...",
+            f"{solver_key}/releases/{version}/jobs/{job_id}/outputs/logfile",
+            presigned_download_link,
+        )
+        return RedirectResponse(presigned_download_link)
```

Comment:
I guess this only replies with 307, not with a 200 as well?

Reply:
As explained above